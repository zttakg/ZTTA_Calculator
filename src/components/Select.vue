<template>
  <span :class="classes">
    <select
      class="form-control form-control_type_select select__control"
      :id="name"
      :name="name"
      :value="value"
      :multiple="multiple"
      :disabled="disabled"
      @input="handleInput"
    >
      <option
        v-if="placeholder"
        class="select__placeholder"
        value=""
        :selected="value === '' ? 'selected' : false"
        disabled
      >
        {{ placeholder }}
      </option>
      <option
        v-if="hasOptions"
        v-for="option in options"
        :key="option.value"
        :value="option.value"
        :label="option.label"
        :disabled="option.disabled"
        :selected="option.value === value ? 'selected' : false"
        :hidden="option.hidden"
      >
        {{ option.label }}
      </option>
      <slot v-if="!hasOptions"></slot>
    </select>
  </span>
</template>

<script>
export default {
  name: "Select",

  props: {
    value: {
      default: 0
    },
    name: String,
    theme: {
      type: String,
      default: "default"
    },
    options: {
      type: Array,
      default() {
        return [];
      }
    },
    placeholder: String,
    disabled: Boolean,
    multiple: {
      type: Boolean,
      default: false
    },
    block: Boolean
  },

  computed: {
    classes() {
      const themes = this.theme
        .split(" ")
        .filter(t => t !== "")
        .map(t => `select_style_${t}`);

      return [
        "select",
        {
          [`select_type_${this.type}`]: this.type,
          select_block: this.block,
          select_inline: !this.block
        },
        ...themes
      ];
    },
    hasOptions() {
      return this.options.length > 0;
    }
  },

  methods: {
    handleInput(event) {
      this.$emit("input", event.target.value);
    }
  }
};
</script>
