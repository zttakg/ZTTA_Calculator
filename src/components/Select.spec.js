import { shallowMount } from "@vue/test-utils";
import Select from "./Select";

describe("<Select /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(Select);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(Select, {
      propsData: {
        value: "2",
        name: "test-name",
        theme: "material rounded",
        options: [
          {
            label: "Item 1",
            value: "1"
          },
          {
            label: "Item 2",
            value: "2"
          },
          {
            label: "Item 3",
            value: "3"
          }
        ],
        placeholder: "Please select ...",
        multiple: true,
        block: true,
        disabled: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render with slot", () => {
    const wrapper = shallowMount(Select, {
      propsData: {
        value: "2",
        name: "test-name",
        theme: "material rounded",
        placeholder: "Please select ...",
        multiple: true,
        block: true,
        disabled: true
      },
      slots: {
        default: [
          {
            label: "Item 1",
            value: "1"
          },
          {
            label: "Item 2",
            value: "2"
          },
          {
            label: "Item 3",
            value: "3"
          }
        ].map(
          option => `<option value="${option.value}">${option.label}</option>`
        )
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
