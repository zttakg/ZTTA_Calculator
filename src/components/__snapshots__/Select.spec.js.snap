// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> component should render all props 1`] = `
<span class="select select_block select_style_material select_style_rounded"><select multiple="multiple" id="test-name" name="test-name" disabled="disabled" class="form-control form-control_type_select select__control"><option value="" disabled="disabled" class="select__placeholder">
      Please select ...
    </option> <option label="Item 1" value="1">
      Item 1
    </option><option label="Item 2" value="2">
      Item 2
    </option><option label="Item 3" value="3">
      Item 3
    </option> <!----></select></span>
`;

exports[`<Select /> component should render correct without props 1`] = `<span class="select select_inline select_style_default"><select class="form-control form-control_type_select select__control"><!---->  </select></span>`;

exports[`<Select /> component should render with slot 1`] = `
<span class="select select_block select_style_material select_style_rounded"><select multiple="multiple" id="test-name" name="test-name" disabled="disabled" class="form-control form-control_type_select select__control"><option value="" disabled="disabled" class="select__placeholder">
      Please select ...
    </option>  <option value="1">Item 1</option><option value="2">Item 2</option><option value="3">Item 3</option></select></span>
`;
